core/listener.go

```diff
@@ -155,19 +162,32 @@ func (cl *Listener) handleNewSignedBlock(ctx context.Context, b types.EventDataS
 	adder := ipld.NewProofsAdder(int(b.Data.SquareSize))
 	defer adder.Purge()
 
-	eds, err := extendBlock(b.Data, b.Header.Version.App, nmt.NodeVisitor(adder.VisitFn()))
-	if err != nil {
-		return fmt.Errorf("extending block data: %w", err)
+	var (
+		dataSq *rsmt2d.ExtendedDataSquare
+		err    error
+	)
+
+	if !app.IsEmptyBlock(b.Data, b.Header.Version.App) {
+		dataSq, err = cl.edsConstruct(
+			b.Data.Txs.ToSliceOfBytes(),
+			b.Header.Version.App,
+			appconsts.SquareSizeUpperBound(b.Header.Version.App),
+			nmt.NodeVisitor(adder.VisitFn()),
+		)
+		if err != nil {
+			panic(fmt.Errorf("extending block data: %w", err))
```

Comment:
Is it really necessary to crash here? Why not just return an error as it used to be?